### ISSUE TITLE
[crm] Incorrect crm counter for ipv4(6)_next_hop 

### ISSUE THREAD

**Issue by bingwang-ms** [Azure/sonic-buildimage#5583]
**Description**
The crm counter for ipv4_next_hop and ipv6_next_hop is incorrect, which leads to test failure in test_crm.

**Steps to reproduce the issue:**
1. Set crm polling interval to 1 second ```crm config polling interval 1```.
 The default polling interval is 300 seconds. This setting is to reduce waiting time between set and check. 
2. Retrieve used and available nexthop count
```
root@str-dx010-acs-4:~# redis-cli --raw -n 2 HMGET CRM:STATS crm_stats_ipv4_nexthop_used crm_stats_ipv4_nexthop_available
4
32691
```
3. Add a permanent nexthop
```
root@str-dx010-acs-4:~# ip neigh replace 2.2.2.2 lladdr 11:22:33:44:55:66 dev PortChannel0001
``` 
4. Retrieve new value of used and available nexthop count after several seconds.
```
root@str-dx010-acs-4:~# redis-cli --raw -n 2 HMGET CRM:STATS crm_stats_ipv4_nexthop_used crm_stats_ipv4_nexthop_available
5
32689
```
The used count is increased by 1, while the available count is decreased by 2
5. Delete the nexthop set in step 3
```
ip neigh del 2.2.2.2 lladdr 11:22:33:44:55:66 dev PortChannel0001
```
6. Retrieve new value of used and available nexthop count after deleting nexthop after several seconds.
```
root@str-dx010-acs-4:~# redis-cli --raw -n 2 HMGET CRM:STATS crm_stats_ipv4_nexthop_used crm_stats_ipv4_nexthop_available
4
32690
```
The used count of nexthop is restored to 4. However, the available count is not recoverd. This issue also exists for ipv6_nexthop.

**Describe the results you received:**
The available count of nexthop is not recoverd after deleting manually added nexthop. 

**Describe the results you expected:**
The available count of nexthop is exactly same after deleting manually added nexthop with that before adding the nexthop. 

**Additional information you deem important (e.g. issue happens only occasionally):**

    **Output of `show version`:**

```
SONiC Software Version: SONiC.master.436-d03de95e
Distribution: Debian 10.6
Kernel: 4.19.0-9-2-amd64
Build commit: d03de95e
Build date: Wed Oct  7 17:20:56 UTC 2020
Built by: johnar@jenkins-worker-11

Platform: x86_64-cel_seastone-r0
HwSKU: Celestica-DX010-C32
ASIC: broadcom
Serial Number: DX010F2B118711MS100007
Uptime: 10:29:26 up  4:54,  1 user,  load average: 2.05, 2.14, 2.05

Docker images:
REPOSITORY                    TAG                   IMAGE ID            SIZE
docker-teamd                  latest                241aa411e774        394MB
docker-teamd                  master.436-d03de95e   241aa411e774        394MB
docker-nat                    latest                bf3f23c5234e        396MB
docker-nat                    master.436-d03de95e   bf3f23c5234e        396MB
docker-router-advertiser      latest                b9fe986fb925        359MB
docker-router-advertiser      master.436-d03de95e   b9fe986fb925        359MB
docker-platform-monitor       latest                e269c5ce7efa        441MB
docker-platform-monitor       master.436-d03de95e   e269c5ce7efa        441MB
docker-lldp                   latest                04e472d56518        388MB
docker-lldp                   master.436-d03de95e   04e472d56518        388MB
docker-database               latest                c8f685098ffa        359MB
docker-database               master.436-d03de95e   c8f685098ffa        359MB
docker-orchagent              latest                7b06e0aec659        407MB
docker-orchagent              master.436-d03de95e   7b06e0aec659        407MB
docker-dhcp-relay             latest                c0f95ee6d5ae        366MB
docker-dhcp-relay             master.436-d03de95e   c0f95ee6d5ae        366MB
docker-sonic-telemetry        latest                dda93d38bfc9        429MB
docker-sonic-telemetry        master.436-d03de95e   dda93d38bfc9        429MB
docker-sonic-mgmt-framework   latest                822459bae952        486MB
docker-sonic-mgmt-framework   master.436-d03de95e   822459bae952        486MB
docker-sflow                  latest                f0c87436a533        397MB
docker-sflow                  master.436-d03de95e   f0c87436a533        397MB
docker-fpm-frr                latest                6e80b427ec49        410MB
docker-fpm-frr                master.436-d03de95e   6e80b427ec49        410MB
docker-snmp                   latest                4f48e26b7dee        399MB
docker-snmp                   master.436-d03de95e   4f48e26b7dee        399MB
docker-syncd-brcm             latest                2198e1e38cfb        542MB
docker-syncd-brcm             master.436-d03de95e   2198e1e38cfb        542MB
```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by gechiang** [comment]
This looks to be new issue introduced by SAI 4.2.1.3
I rolled back to previous version (3.7.5.1) and it was working fine.
On the very first version of master image using SAI 4.2.1.3 this issue is observed.
Tried on two different BRCM DUTs  one on Celestica-DX010-C32 (BCM56960_B1 ) and another one on Force10-S6000 (BCM56850_A2).  Both shows the same failing behavior.
From the code this available resource counter is read from SAI via sai_switch_api->get_switch_attribute with attribute "SAI_SWITCH_ATTR_AVAILABLE_IPV4_NEXTHOP_ENTRY" (IPV4 case) and somehow this number is being decremented by 2 instead of by 1 when a new nexthop is used.  When the nexthop is not used, this resource counter got restored back by 1 only.
Have opened a new BRCM case (CS00011396506) requesting BRCM to investigate/address this issue.
